apps/internal/base/base.go

```diff
@@ -280,26 +280,8 @@ func (b Client) AuthCodeURL(ctx context.Context, clientID, redirectURI string, s
 }
 
 func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilentParameters) (AuthResult, error) {
+	// when tenant == "", the caller didn't specify a tenant and WithTenant will use the client's configured tenant
 	tenant := silent.TenantID
-	if tenant == "" {
```

Comment:
Sure looks simpler :)

Reply:
😆 it's definitely that. For reference, here's [`AuthParams.WithTenant()`](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/8531f217b6e983184cab9bba88b80600ea879d2b/apps/internal/oauth/ops/authority/authority.go#L183), which sets the tenant for each authentication and uses the client's configured tenant when the caller doesn't specify an override.